### DISCUSSION TITLE
Running moveit on aarch64 ros 2 humble

### DISCUSSION THREAD

**Discussion by rebop3000** [moveit/moveit2#2558] [Q&A]
colcon build is unable to build the kortex api and kortex drivers because they are for a different architecture, 84x something, i can skip it in the build but then I can't run any of the tutorials because its missing issac joint angles. I am wondering if I can even use moveit 2 with my configuration and is there anyway to get this to work? Thanks so much!

**Comment by okvik** [comment]
We're using MoveIt2 on Humble on aarch64 machines, in production even.

**Reply by BillyMazotti** [reply]
Do you have any resource recommendations?

**Reply by okvik** [reply]
Resources for what in particular?

For running on aarch64 there's really nothing special to note, it just worked on our machines. Granted, we had to rent a potent ARM64 build machine that could build MoveIt2 from source without eating all the RAM and taking forever.

If you're asking about resources on using MoveIt2 in general, there's unfortunately not much I can suggest. If the official docs aren't enough you'll pretty much have to go read the code to figure how things work (or don't). Also, read how others are using the library by searching for code here on Github.

Good luck.

**Reply by BillyMazotti** [reply]
Thank you for the response! I ended up rechecking the official docs and realized I goofed up again.

Official docs ended up being enough, however lack of experience with specificity and explicitness (e.g. MoveIt's docs vary for [humble](https://moveit.picknik.ai/humble/doc/tutorials/getting_started/getting_started.html) vs [iron](https://moveit.picknik.ai/main/doc/tutorials/getting_started/getting_started.html)) lead to wasted time albeit with opportunity for many lessons learned (e.g. I should've tried to see if MoveIt had a specific page for humble rather than trying to copy and past the iron page).

**Comment by MBenediktF** [comment]
I'm running into the same issue. When I try to compile the workspace following the [official documentation](https://moveit.picknik.ai/main/doc/tutorials/getting_started/getting_started.html), I get the same CMake error:
```--- stderr: kortex_api                                               
CMake Error at CMakeLists.txt:26 (message):
  Unsupported System: currently support is for Linux x68_64.  Detected Linux
  and aarch64


---
```
I'm using Ros2 Iron Irwini on my Ubuntu22.04 virtual machine, running via UTM on my M1 Mac.

Is there an easy solution?

**Reply by tylerjw** [reply]
Just take the kortex drivers out of your workspace and ignore the packages in the rosdep step and it should work. The tutorials should still work, you just won't be able to run with a physical kanova arm (as they put closed source x86 binaries in their driver).

**Reply by BillyMazotti** [reply]
Following @tylerjw 's advice and using [Moveit's Getting Started steps for ROS Iron](https://moveit.picknik.ai/main/doc/tutorials/getting_started/getting_started.html), I removed the package ros2_kortex after running "vcs import..." and before building the colcon workspace.

It appears that kortex is not apart of the [MoveIt's Getting Started steps for humble](https://moveit.picknik.ai/humble/doc/tutorials/getting_started/getting_started.html) which I suppose eliminates the need for removing the kortex package as long as you correctly specify the moveit2_tutorials branch.

These steps were only tested with iron, not humble, however they should work to the best of my knowledge.

For Humble
```
cd ~/ws_moveit/src
git clone --branch humble https://github.com/ros-planning/moveit2_tutorials
vcs import < moveit2_tutorials/moveit2_tutorials.repos
sudo apt update && rosdep install -r --from-paths . --ignore-src --rosdistro $ROS_DISTRO -y
cd ~/ws_moveit
colcon build --mixin release
```

For Iron
```
cd ~/ws_moveit/src
git clone https://github.com/ros-planning/moveit2_tutorials
vcs import < moveit2_tutorials/moveit2_tutorials.repos
rm -rf ~/ws_moveit/src/ros2_kortex. 
sudo apt update && rosdep install -r --from-paths . --ignore-src --rosdistro $ROS_DISTRO -y
cd ~/ws_moveit
colcon build --mixin release
```

Removing the kortex package does mean you won't be able to run the Kinova arm which is the suggested demo arm since I believe kortex is required based on [ros_kortex's repo](https://github.com/Kinovarobotics/ros_kortex).

I'm workin on Macbook Pro M1 running Ubuntu 22.04 using VMware Fusion. I was having difficulty with getting MoveIt setup properly for Humble however that may have been because I was following the ROS Iron MoveIt Getting Started rather than the ROS Humble MoveIt Getting Started steps. In any case Iron seems to be working. I did not try to see if Humble steps work for humble, at a quick glance the major difference in steps seems to be cloning the humble branch of moveit2_tutorials.

I was able to simulate the Panda demo robot arm for Iron without any issues using:
`
ros2 launch moveit_resources_panda_moveit_config demo.launch.py
`
